lm_scorer/models/gpt2.py

```diff
@@ -27,47 +36,45 @@ def _add_special_tokens(self, text: str) -> str:
     def _tokens_log_prob_for_batch(
         self, text: List[str]
     ) -> List[Tuple[torch.DoubleTensor, torch.LongTensor, List[str]]]:
-        device = self.model.device
-
-        outputs = []
-        for sentence in text:
-            # len(sentence) = seq_len + 2
-            sentence = self._add_special_tokens(sentence)
-            # len(tokens) = seq_len + 2
-            tokens: List[str] = self.tokenizer.tokenize(sentence)
-            # ids.shape = [1, seq_len + 2]
-            ids = torch.tensor(  # pylint: disable=not-callable
-                [self.tokenizer.convert_tokens_to_ids(tokens)],
-                device=device,
-                dtype=torch.long,
-            )
-
-            with torch.no_grad():
-                model_outputs: Tuple[torch.Tensor] = self.model(ids)
-
-            # pred_scores.shape = [1, seq_len + 2, vocab_size]
-            pred_scores = model_outputs[0].double()
-
-            # len(tokens) = seq_len + 1
-            tokens = tokens[1:]
-            # ids.shape = [1, seq_len + 1, vocab_size]
-            ids = ids[:, 1:]
-            # pred_scores.shape = [1, seq_len + 1, vocab_size]
-            pred_scores = pred_scores[:, :-1, :]
-
-            # ids_scores.shape = [1, seq_len + 1]
-            ids_scores = pred_scores.gather(2, ids.unsqueeze(2)).squeeze(2)
-            # log_prob.shape = [1, seq_len + 1]
-            log_probs = ids_scores - pred_scores.logsumexp(2)
-
-            log_probs = log_probs.squeeze(0)
-            ids = ids.squeeze(0)
-
-            log_probs = cast(torch.DoubleTensor, log_probs)
-            ids = cast(torch.LongTensor, ids)
-
-            output = (log_probs, ids, tokens)
+        outputs: List[Tuple[torch.DoubleTensor, torch.LongTensor, List[str]]] = []
+        if len(text) == 0:
+            return outputs
+
+        # TODO: Handle overflowing elements for long sentences
```

Comment:
If you have time, would you mind looking into this?

The `batch_encode_plus` has different strategies to handle the overflowing tokens when the model has a maximum sequence lenght.

What we would have to do is:
- [ ] Add a test with a sentence longer than the model max sequence lenght.
- [ ] Split the sentence in multiple pieces during processing and then put the pieces toghether.

We can also only add a failing test for us to remember it and mark it as "expected to fail" with the `@pytest.mark.xfail` annotation.

Reply:
We can also just truncate the string and log a warning to the user if the sentence is too long.

Reply:
I think the best splitting strategy can be very domain specific and is not trivial. 
For instance if have a big text of let says 3000 tokens. I don't think the best way is to split the sentence in part for instance in 3*1000 tokens (or 1024 + 1024 + ...) compute the score of each part independently and then concatenate the results.  Because when doing so the probability associated with token n°1025 will be very wrong. 
So it would probably better to have sliding windows. The best would be to move one by one  so that when computing the prob of a tokens you take into account the largest context you can. However this would be very computationally demanding and not suitable for every user but we could solved this problem by giving the possibility  to the users to choose the sliding step. 
So, in my opinion, the best for now is to simply raise an error (or truncate + warning) and works latter on this topic on an other branch.

Reply:
The `batch_encode_plus` has a `stride` argument to set how many tokens from the previous "part" should be used in the next "part" when splitting.

Anyway I agree that we can do that later.
Let's just add the test then (the current behaviour should already be that it truncates long sentences, but let's verify that with a test).

Reply:
For now, it raises an RuntimeError (index out of range error). 
```
    def should_throw_an_exception_for_too_large_sentence():
        max_input_size = scorer.model.config.n_positions
        long_sentence = ["The " * max_input_size]
        with pytest.raises(RuntimeError):
            scorer._tokens_log_prob_for_batch(long_sentence)
```
We just add this test ?

Reply:
Let's create a test that tests the correct behaviour and let's mark it as "expected to fail".
```python
@pytest.mark.xfail
def should_work_with_a_sentence_longer_than_the_model_max_size():
        max_input_size = scorer.tokenizer.max_len
        long_sentence = ["The " * (max_input_size + 1)]
        scorer._tokens_log_prob_for_batch(long_sentence)
        # Add asserts that describe how it should behave if we were to correctly implement it
```